src/lib/warn-upload-overwrite.js

```diff
@@ -0,0 +1,94 @@
+const fs = require('./sync-fs');
+const jsonDiff = require('json-diff');
+const readline = require('readline-sync');
+const crypto = require('crypto');
+
+const question = 'You are trying to modify a configuration that has been modified since your last upload. Do you want to?';
+const responseChoicesWithoutDiff = [
+  'Overwrite the changes', 
+  'Abort so that you can update the configuration'
+];
+const responseChoicesWithDiff = responseChoicesWithoutDiff.concat([ 'View diff' ]);
+
+const preUpload = async (projectDir, db, doc, couchUrl) => {
+  let localDoc = JSON.parse(JSON.stringify(doc));
+
+  // Pull remote _rev
+  let remoteRev;
+  let remoteDoc;
+  try {
+    remoteDoc = await db.get(localDoc._id);
+    remoteRev = remoteDoc._rev;
+  } catch (e) {
+    // continue regardless of error
+  }
+
+  // Pull local _rev
+  let localRev;
+  try {
+    const md5 = crypto.createHash('md5').update(couchUrl).digest('hex');
+    localRev = fs.read(`${projectDir}/._revs/${md5}/${localDoc._id}`).trim();
```

Comment:
Makes sense. The only downside is if you have a git conflict you won't be able to tell which server you conflicted on without reversing the hash.

The other way would be to have a single doc at `${projectDir}/_revs/${localDoc._id}.json` which has a map from `couchUrl` to `_rev`. This would mean you don't need to hash the URL.

You could take that one step further and have ``${projectDir}/.revs.json` which has a map of `couchUrl` to `localDoc._id` to `_rev` which might make file access a little simpler so reads and writes are easy. The downside being you're more likely to conflict...

I'll leave it up to you to decide.